### PR TITLE
VFB-167: Display additional fields in the users table on the admins page

### PR DESCRIPTION
**Pull request by minalui**
## What's changed
- Display additional user profile information: First name, last name, telephone number, user role
- Allow filtering and sorting

## Screenshots / Videos
| Before     | After      |
|------------|------------|
| image | image |

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA
- [x] I have self-reviewed this PR
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
- [x] Make sure you've tested via `npm run test:e2e`

If you have made any changes to the database...
  - [x] The migration files are up-to-date with my final set up (`npx supabase db diff -f <name_of_migration>` should create nothing at this point)
  - [x] I have updated the typescript definitions for the database with `db:local:generate_types`
  - [N/A] I have modified the seed in `supabase/seed/seed.mts` if appropriate 
  - [x] If I have modified the seed, I have also generated the seed with `npm run db:generate_seed` 
  - [x] With my final set up, I can run `npm run dev:reset_supabase` without any errors.
  - Does this require resetting the deployed database? - YES (remove as appropriate)
  - [x] I have checked that migration can happen successfully without resetting the database, doing the following.
    - Make sure you have rebased your branch onto dev or merged dev into your branch.
    - Checkout the dev branch
    - Run `npm run post_checkout` to reset the database, including the seed data.
    - Checkout your branch
    - Run `npx supabase migrations up --include-all --local` to run all outstanding migration files
    - Check that the resulting database is what you expect.

### CODE REVIEW

File: src/app/admin/usersTable/getUsersData.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    const userData: UserRow[] = [];
+    let query = supabase.from("profiles_plus").select("*");
+
+    if (
+        sortState.sortEnabled &&
+        sortState.column.sortMethodConfig?.paginationType === PaginationType.Server
+    ) {
+        query = sortState.column.sortMethodConfig.method(query, sortState.sortDirection);
+    } else {
+        query = query.order("first_name");
```
Comment (minalui) [inline src/app/admin/usersTable/getUsersData.ts:25]: Should we order by first name or last name?

**Review by rickaw — CHANGES_REQUESTED** [review]

File: supabase/migrations/20240409100702_add_profile_plus_view.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+create or replace view "public"."profiles_plus" as  SELECT profiles.primary_key AS user_id,
+    profiles.first_name,
+    profiles.last_name,
+    profiles.role,
+    profiles.telephone_number,
+    users.email,
+    users.created_at,
+    users.updated_at
+   FROM (profiles
+     LEFT JOIN auth.users ON ((profiles.primary_key = users.id)))
```
[the remark this answers is not in the document]
Comment (minalui) [reply supabase/migrations/20240409100702_add_profile_plus_view.sql:10]: Yep 💯

File: src/app/clients/getClientsData.ts [hunk 3]
```diff
@@ -41,44 +67,38 @@ const getClientsDataAndCount = async (
     const { data: clients, error: clientError } = await query;
 
     if (clientError) {
-        const logId = abortSignal.aborted
-            ? await logInfoReturnLogId("Aborted fetch: client table", clientError)
-            : await logErrorReturnLogId("Error with fetch: client table", clientError);
         if (abortSignal.aborted) {
-            throw new AbortError("fetch", "client table", "logId");
+            const logId = await logInfoReturnLogId("Aborted fetch: client table", clientError);
```
Comment (rickaw) [inline src/app/clients/getClientsData.ts:71]: (same in other places please)
```suggestion
            const logId = await logInfoReturnLogId("Aborted fetch: client table", { error: clientError });
```

File: src/app/clients/getClientsData.ts [hunk 4]
```diff
@@ -41,44 +67,38 @@ const getClientsDataAndCount = async (
...
-        });
-    }
+        };
+    });
 
-    const count = await getClientsCount(supabase, filters, abortSignal);
+    const countData = await getClientsCount(supabase, filters, abortSignal);
+    if (countData.data === null) {
+        return countData;
+    }
```
Comment (rickaw) [inline src/app/clients/getClientsData.ts:91]:
```suggestion
    const{ data: clientCount, error: clientCountError } = await getClientsCount(supabase, filters, abortSignal);
    if (clientCountError) {
        return { data: null, error: clientCountError };
    }
```

File: src/app/clients/getClientsData.ts [hunk 5]
```diff
@@ -41,44 +67,38 @@ const getClientsDataAndCount = async (
...
+    console.log("here");
```
Comment (rickaw) [inline src/app/clients/getClientsData.ts:92]:
```suggestion
```

File: src/common/fetch.ts [hunk 6]
```diff
@@ -172,3 +180,21 @@ export const fetchPackingSlotsInfo = async (
...
+    supabase: Supabase
+): Promise<UserProfileDataAndError> => {
+    const { data, error } = await supabase
+        .from("profiles")
+        .select("role, first_name, last_name, telephone_number")
+        .eq("user_id", userId)
+        .single();
+
+    if (error) {
+        void logErrorReturnLogId("Failed to fetch: user profile", error);
```
Comment (rickaw) [inline src/common/fetch.ts:195]:
```suggestion
        void logErrorReturnLogId("Failed to fetch: user profile", { error });
```

File: src/app/admin/usersTable/UsersTable.tsx [hunk 7]
```diff
@@ -27,76 +98,160 @@ const formatTimestamp = (timestamp: number): string => {
...
+
+            if (error) {
+                switch (error.type) {
+                    case "abortedFetchingProfilesTable":
+                    case "abortedFetchingProfilesTableCount":
+                        return;
+                    case "failedToFetchProfilesTable":
+                    case "failedToFetchProfilesTableCount":
+                        setErrorMessage(`Error occurred: ${error.type}, Log ID: 
+                    ${error.logId}`);
```
[the remark this answers is not in the document]
Comment (minalui) [reply src/app/admin/usersTable/UsersTable.tsx:209]: Yes forgot this!

**Review by rickaw — APPROVED** [review]

**Pull request merged**